contracts/WrapperRegistry.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.4;
+
+import {IWrapperRegistry} from "./interfaces/IWrapperRegistry.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+/**
+ * @title The Wrapper Registry
+ *
+ * @dev Registers bijective pairs of tokens and button/unbutton wrappers
+ *
+ */
+contract WrapperRegistry is IWrapperRegistry, Ownable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /**
+     * @dev Mapping of underlying tokens to wrapper tokens
+     */
+    mapping(address => address) private _wrapperMapping;
+    /**
+     * @dev Enumerable set of underlying token addresses
+     * @dev Required since EnumerableMap only supports uint256->address maps
+     */
+    EnumerableSet.AddressSet private _underlyingTokens;
+
+    /**
+     * @inheritdoc IWrapperRegistry
+     * @dev Stores address of underlyingToken into `_underlyingTokens`
+     * @dev Stores an entry of underlyingToken address to wrapperToken address into `_wrapperMapping`
+     */
+    function addWrapper(address underlyingToken, address wrapperToken)
```

Comment:
You just need one parameter here, ie the wrapper token. You can query the underlying token using `IWrapper(wrapperToken).underlying()` rather than taking it as input from the user. This will remove potential data inconsistency from passing the wrong addresses in.

Alternatively if you prefer two parameters you should add an assertion to check if the underlying token passed to the function address matches with the wrapper's underlying token address